### DISCUSSION TITLE
Proper management of dynamically updated vector database associated with a RAG agent

### DISCUSSION THREAD

**Discussion by rrrozhd** [microsoft/autogen#2440] [Q&A]
Hi there! Can someone advise me on what should I be weary of when implementing the following architecture. Let's say I need one of my agents to retrieve an API documentation from the web and save it to the local vector database, and the other one to write and execute an API request based on that documentation. 

Suppose there is a tool the first agent can call that downloads a webpage and saves it to the vector database. Suppose the second agent is a RAG agent that accesses the same vector database afterwards for retrieving relevant information (for instance which endpoint is best suited for the user task) for writing its request. 

Are there any significant fallacies in this general sort of design? 
For instance, would I have to reinitialize the RAG agent every time the database is updated? 

I am new to this framework so would be grateful for someone to shed some more light on this!

**Comment by sonichi** [comment]
@thinkall fyi

**Comment by thinkall** [comment]
> Hi there! Can someone advise me on what should I be weary of when implementing the following architecture. Let's say I need one of my agents to retrieve an API documentation from the web and save it to the local vector database, and the other one to write and execute an API request based on that documentation.
> 
> Suppose there is a tool the first agent can call that downloads a webpage and saves it to the vector database. Suppose the second agent is a RAG agent that accesses the same vector database afterwards for retrieving relevant information (for instance which endpoint is best suited for the user task) for writing its request.
> 
> Are there any significant fallacies in this general sort of design? For instance, would I have to reinitialize the RAG agent every time the database is updated?
> 
> I am new to this framework so would be grateful for someone to shed some more light on this!

Hi @turgor111 , with our current design of vector db, you don't need to reinitialize the RAG agent.

**Comment by thinkall** [comment]
Close as it's not active for a long time. Please reopen if the issue still persist.
